### DISCUSSION TITLE
Usage of environment variables in a project based on rspack/rsbuild

### DISCUSSION THREAD

**Discussion by Dev-mininos** [web-infra-dev/rsbuild#2772] [Q&A]
Hello.
I have been trying to migrate my project from vite to rspack/rsbuild and everything works smoothly except i have a little issue concerns how to use the environment variables, i read the documentation about this subject and i tried a lot of methods but none of them seems to work
any solutions?

**Comment by sysmat** [comment]
I have the same problem

- vite & .env.prod

```
const env = loadEnv(mode, process.cwd());
  return {
    base: env.VITE_BASE_PATH || '/',
```

- index.tsx

```
const basePath = import.meta.env.VITE_BASE_PATH || '/';
```

**Comment by YanPes** [comment]
@Dev-mininos what is the exact issue you face?
